sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/expressions/expressions.scala

```diff
@@ -183,17 +149,10 @@ private[sql] final case class LiteralValue[T](value: T, dataType: DataType) exte
 }
 
 private[sql] final case class FieldReference(parts: Seq[String]) extends NamedReference {
+  import org.apache.spark.sql.catalog.v2.CatalogV2Implicits._
```

Comment:
Do we only need `CatalogV2Implicits` for `quoted`?

Reply:
Yes.

Reply:
I don't think we should import everything from `CatalogV2Implicits` then. Importing implicits can invoke unexpected calls to said implicit methods.

Reply:
Can you be more specific about what the risk is?

Reply:
My understanding of Scala implicits is that the implicit conversions or methods are exactly that - implicit - so they can be called without specifically invoking the method names. So a common philosophy around implicits is to only import them if such conversions are definitely required. Otherwise we can create side effects of calling the implicit methods when we don't mean to.

Reply:
This makes all of the implicit conversions available, but does not apply them unless it is necessary. For these implicit classes, the only time it is necessary is when one of the methods, like `quoted` is called. (The compiler would also implicitly convert when an object is coerced to the implicit type as well, but there are no methods that require an `IdentifierHelper`.)

It is fine to import unused implicits. The cases where the compiler will use them are predictable, and they can be used to provide a richer Scala API for classes that are limited because they are part of the public API.

I'm also wary of using Scala's `implicit`, but the ability to add methods to existing classes is one of the cases where I find them reliable and useful.

Reply:
I still think that here we should only be importing the specific method that's used - maybe move `quoted` to a helper class. If we really need the implicits functionality we can import it later on. Can we do that?

Reply:
I can import just the required implicit class, but I don't think there is any risk to using implicits like this.

Reply:
I've updated this.